### PR TITLE
#572 Slack notification when new student apply to the course

### PR DESCRIPTION
**Pull request by aYaremchuk**
Resolves [github issue](https://github.com/pivorakmeetup/pivorak-web-app/issues/x)

### Description

Describe the changes and motivations for the pull request, unless obvious from the title.

### Pre-merge checklist
 
- [ ] The PR relates to a single subject with a clear title and description
- [ ] Verify that feature branch is up-to-date with `development` (if not - rebase it). 

### Screenshots

| Before                                      | After                                       |
| ------------------------------------------- | ------------------------------------------- |
| Insert screenshots and/or screen recordings | Insert screenshots and/or screen recordings |

### CODE REVIEW

File: components/courses/app/controllers/courses/students_controller.rb [hunk 1]
```diff
@@ -10,7 +10,9 @@ def new
 
     def create
       @student = Student::Create.call(student_params, current_season.id, current_user.id)
-      react_to student.save
+      student_creation_status = student.save
+      NotifySlack::Base.new(student.id).call if student_creation_status
```
Comment (sveredyuk) [inline components/courses/app/controllers/courses/students_controller.rb:14]: We should not strict to slack. Tomorrow we will change Slack to SomeOtherNewFashianableChat (Telegram) and we have to rewrite code across all app.

Better solution: create adapter `Notifier`. And it should know what and how to do.

File: lib/notify_slack/base.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      @notifier = Slack::Notifier.new(ENV['SLACK_WEBHOOK_URL'],
+                                      channel: ENV['SLACK_CHANNEL'],
+                                      username: ENV['SLAsCK_USERNAME'])
+      @student_id = student_id
+    end
+
+    attr_accessor :notifier, :student_id
+
+    def call
+      student = Courses::Student.find(student_id)
```
Comment (sveredyuk) [inline lib/notify_slack/base.rb:13]: Unfortunately coupling main app with Courses engine (and Students) is impossible.

**Review by sveredyuk — COMMENTED** [review]
Good initiative, but need to do a lot of fixes

File: lib/notify_slack/base.rb (hunk 2 above)
Comment (sveredyuk) [reply lib/notify_slack/base.rb:13]: I mean, can we move this code into `courses` engine?

**Comment by skrix** [comment]
Closing in cause of inactivity for more than 1 year.

**Pull request closed without merging**